src/screens/SettingsScreen.tsx

```diff
@@ -56,47 +42,117 @@ export const SettingsScreen = () => {
           />
-          <View style={styles.subMenuContainer}>
-            <OAuthLogoutButton label={t('settings-logout', 'Logout')} />
-            <View style={styles.versionContainer}>
-              <Text testID={tID('version-text')}>
-                {t('settings-version', 'Version {{ version }}', {
-                  version: fullVersion,
-                })}
-              </Text>
-            </View>
-          </View>
+          <Divider />
         </ScrollView>
+        <View style={styles.subMenuContainer}>
+          <OAuthLogoutButton label={t('settings-logout', 'Logout')} />
+          <View style={styles.versionContainer}>
+            <Text testID={tID('version-text')}>
+              {t('settings-version', 'Version {{ version }}', {
+                version: fullVersion,
+              })}
+            </Text>
+          </View>
+        </View>
       </SafeAreaView>
     </View>
   );
 };
 
-const styles = StyleSheet.create({
-  mainMenuItem: {
+interface Props {
+  title: string;
+  action?: () => void;
+  badge?: React.ReactNode;
+}
+
+function MainMenuItem({ title, action, badge }: Props) {
+  const { styles } = useStyles('SettingsScreen', defaultStyles);
+
+  return (
+    <TouchableOpacity onPress={action} accessibilityRole="button">
+      <View style={styles.mainMenuItem}>
+        <Text style={styles.mainMenuItemText}>{title}</Text>
+        {badge}
+        {
+          // TODO: Replace this with default actual icons. Possibly from react native paper.
+        }
+        <Text style={styles.arrow}>{'>'}</Text>
+      </View>
+    </TouchableOpacity>
+  );
+}
+
+function Divider() {
+  const { styles } = useStyles('SettingsScreen', defaultStyles);
+
+  return <View style={styles.divider} />;
+}
+
+const defaultStyles = (theme: Theme) => {
+  const mainMenuItem: ViewStyle = {
     flex: 1,
     flexDirection: 'row',
     paddingVertical: 14,
     justifyContent: 'space-between',
-  },
-  subMenuContainer: {
+  };
+
+  const mainMenuItemText: TextStyle = {
+    color: theme.colors.primary,
+  };
+
+  const subMenuContainer: ViewStyle = {
     flex: 1,
-    paddingVertical: 14,
-    paddingHorizontal: 14,
-  },
-  container: {
+    padding: 14,
+    justifyContent: 'center',
+    alignItems: 'center',
+  };
+
+  const arrow: TextStyle = {
+    marginRight: 8,
+    color: theme.colors.outline,
+  };
+
+  const container: ViewStyle = {
     flex: 1,
-  },
-  versionContainer: {
+    marginTop: 8,
+  };
+
+  const versionContainer: ViewStyle = {
     paddingVertical: 14,
     marginTop: 28,
-  },
-  scroll: {
+  };
+
+  const scroll: ViewStyle = {
     flex: 1,
     paddingHorizontal: 28,
-  },
-});
+    flexGrow: 4,
+  };
+
+  const divider: ViewStyle = {
+    height: StyleSheet.hairlineWidth,
+    backgroundColor: theme.colors.outline,
+  };
+
+  return {
+    mainMenuItem,
+    mainMenuItemText,
+    arrow,
+    subMenuContainer,
+    container,
+    versionContainer,
+    scroll,
+    divider,
+  };
```

Comment:
I'm still not in love with this pattern because it forces us to make new variables for every named style, produces lengthy styles for even small style definitions, and is frequently redundant because of definitions like `const fooText: TextStyle = {`

I propose we create a utility function that infers the style types based on property suffixes and allow for providing the namespace in a single place that can be used to extract the namespace in both `ComponentNamedStyles` and the `useStyles` hook.

```typescript
type StringLiteral<T> = T extends `${string & T}` ? T : never;
type IgnoreCapitals<T> = T extends string ? Uncapitalize<T> | Capitalize<T> : never;
type TextSuffixes = 'Text' | 'Label';
type ImageSuffixes = 'Image';

type NamedStyles<T> = {
    [P in keyof T]:
    P extends `${string}${IgnoreCapitals<TextSuffixes>}` ? TextStyle :
    P extends `${string}${IgnoreCapitals<ImageSuffixes>}` ? ImageStyle :
    ViewStyle
}

type CreateStyles = <N extends string, T extends NamedStyles<T>>(
    name: N,
    builder: T | ((theme: Theme) => T)) => { namespace: StringLiteral<N> } & NamedStyles<T>;

const createStyles: CreateStyles = (namespace, builder) => {
    const styles = typeof builder === 'function' ? builder({ theme: 'theme' }) : builder;
    return Object.assign({ namespace }, styles) as any;
}
```

[TS Playground Demo](https://www.typescriptlang.org/play?ts=4.8.4#code/C4TwDgpgBAKgFhAttAvFA3lYDkC4oDk2SEBUAvgNwBQoksEAHsAMqgA2qGWTrHEAZ3wDgAJwCWAOwDmFWuGgA1cRADubEJyhpMANxXr+QqCIky5daAElEAQ2kQNWnVHF2HTwcLFTZ5avL0bGbSADLiwBCituwAPDAAfNqwULwQkgAmAlAABgAk6Ka+UABksOQ5UAD8KfiSELpRNJZQVtKSAPaiEADCtmARMQLxSWgwqczpWSY+5jUAqpIAxv2D7OIAXhAjUAA+UH0DwDGb24lQdQ1NAS0wvCwArgBmT+KMgskEd8xk+wShtgARhB2ARmgpWu5HM9Xu9smgCDZ7KQaIFoAA5WzIDKeYbnHTUKBEqAAbQACq5JFAANYQEAdJ6wAC6uEJxIpaUy2XyhVm0nIBTanW6hzWePuMLeggSFWqDGYngubKJHMmXNyBSKMgF6CFXV6q2O7GGSI8krhMsqNVNjn4SuJUGUak81H8aIO3VskVxyVi6ImkXVWukABoUpzppjsbiRgkABTKqCSLEQfDokOJwEPcTsDJRfDjfZxuPEPCwHAQACU2iSMEr1ZQSUwyeQAjAtiWqagwV84Ui0Ti6KS5FKUCjEBxRhGqKWHUkIigS093qM+B6y9tmg+aDjLcE7c7YazObzogbTcTs-nwBmW-hWAUDKgx9zUW0KARTwey2A4jnZBqF9TzjTBSy7IgKzIchq3wIDrgdbpgAeUQqQAeUBAArCAlmAAA6WwBAEcR2hApMUzbDtoHIMMRDvasCKgWxJBAGg3U4G9IhEAAmZIlwgL1N04AQ4wIJ4Og6Agwzjc8oBAxMxI6b5gHwUD7lXQhIh+MNqPk8SAWBdgVJ4BV1KIXgoIzB12CBEEjM0vg73wMyfgoSziSvY4pHzbh9GdUzfNUCzE0QEAbSMtxkVxJyIocIKHXslTE3itTHMIBSCETfxoMrAJ2J4EQABZeI3XERMBWxREk2T0F0xTeDslKhKc+yqp0h0FP02zuHsqKNPM1zE2sgyGpM1LnOAOL3LnTz6lEIyAt6ggAsmokQrC7gYsErxCE2oLspuQRgC43C9woztaEO46FKU3CeqMZpLtwjqbPYW7GsEB7uNwjzbC80RcIW+6Lq+taoVwzbcVRTjgAKk7yIPCBgZhp7xJuu670+5HnoMt7RqEzHYZ+v6AYMSGkdh0HkXBqFIaAA)

This allows us to create styles in a concise way and allow the name we use to strongly type the property values and produce a return type that can be easily understood by a library consumer.

```typescript
// Styles would now look like this
const defaultStyles = createStyles('Example', (theme) => ({
  label: {
    fontSize: theme.font.body
  },
  image: {
    resizeMode: 'cover'
  },
  headerText: {
    fontSize: 20
  }
});

// as opposed to this
const defaultStyles = (theme) => {
  const label: TextStyle = {
    fontSize: theme.font.body
  };

  const image: ImageStyle = {
    resizeMode: 'cover'
  };

  const headerText: TextStyle = {
    fontSize: 20
  }

  return {
    label,
    image,
    headerText
  };
};
```

Reply:
Love this, I'll create a PR for it.

Reply:
It might be tricky for things like this where we can't control the name though: https://github.com/lifeomic/react-native-sdk/pull/75/files/8f716ccb43959444b3524ab136695f8b8dac2b09#diff-461a3dcd13f11b9d2ca8ac026e59da09377b9f41a4cf34c69c3831ac88e82490R47-R52

I bet there is a way we can make it work though.